components/Navbar/NavbarDropdownComponent.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+import React, { useState } from "react";
+import { Prisma } from "@/lib/prisma";
+
+export type NavbarDropdownProps = Omit<
+  Prisma.NavbarDropdownsGetPayload<{}>,
+  "items"
+> & {
+  items: {
+    title: string;
+    groups_with: string;
+    url: string;
+  }[];
+};
+
+type Item = {
+  title: string;
+  groups_with: string;
+  url: string;
+};
+
+export function NavbarDropdown({ title, items }: NavbarDropdownProps) {
+  const [open, setOpen] = useState(false);
+  const changeState = () => {
+    setOpen(!open);
+  };
+
+  const groupedItems: Item[][] = items.reduce((groups, item) => {
+    if (
+      groups.length === 0 ||
+      groups[groups.length - 1][0].groups_with !== item.groups_with
+    ) {
+      groups.push([item]);
+    } else {
+      groups[groups.length - 1].push(item);
+    }
+    return groups;
+  }, [] as Item[][]);
+
+  return (
+    <>
+      {/* MOBILE */}
+      <div className="md:hidden flex flex-col gap-1">
+        <button
+          onClick={changeState}
+          className="text-brand-yellow text-2xl font-rockingsoda bg-primary  w-full p-3 flex justify-center"
+        >
+          {title}
+        </button>
+        {open && (
+          <div className="pl-2 flex gap-1 flex-col">
+            {groupedItems.map((items) => (
+              <div className="bg-primary w-full p-3 flex flex-col justify-center">
+                <div className="flex w-full gap-2 items-center">
+                  {items[0].groups_with && (
+                    <>
+                      <span className="text-[20px] font-rockingsoda">
+                        {items[0].groups_with}
+                      </span>
+                    </>
+                  )}
+                </div>
+                <div className="w-full h-[1px] bg-brand-yellow"></div>
+                <div className="flex-col flex">
+                  {items.map((item) => (
+                    <a
+                      className="text-center py-2"
+                      href={item.url || undefined}
```

Comment:
Is this ` || undefined` to handle empty strings? Just wondering, no changes needed.